hibernate-core/src/main/java/org/hibernate/engine/internal/Cascade.java

```diff
@@ -141,9 +143,22 @@ else if ( action.requiresNoCascadeChecking() ) {
 							eventSource,
 							parent,
 							persister,
-							types[i],
+							types[ i ],
 							i
 					);
```

Comment:
It's kind of strange having the following block executed when ```action.requiresNoCascadeChecking() == true``` (see line 141).

I'm guessing that it's here because orphans need to be deleted with ```CascadingAction#PERSIST_ON_FLUSH```, even when cascade is not configured.

Can this method be restructured to make this special case stand out better?